aten/src/ATen/native/miopen/Conv_miopen.cpp

```diff
@@ -259,14 +260,17 @@ BenchmarkCache<size_t> bwd_filter_wssizes;
 
 struct Workspace {
   Workspace(size_t size) : size(size), data(NULL) {
-    data = c10::hip::HIPCachingAllocator::raw_alloc(size);
+    data = at::globalContext().userEnabledUVM() ? at::cuda::CachingManagedAllocator::raw_alloc(size) : c10::hip::HIPCachingAllocator::raw_alloc(size);
   }
   Workspace(const Workspace&) = delete;
   Workspace(Workspace&&) = default;
   Workspace& operator=(Workspace&&) = default;
   ~Workspace() {
     if (data) {
-      c10::hip::HIPCachingAllocator::raw_delete(data);
+      if (at::globalContext().userEnabledUVM())
+        at::cuda::CachingManagedAllocator::raw_delete(data);
+      else
+        c10::hip::HIPCachingAllocator::raw_delete(data);
```

Comment:
Is this PR intended to support fine-grained changes of UVM settings, like the following?
```py
torch.cuda.memory.set_enabled_uvm(True)
# [Allocate Tensor1]
torch.cuda.memory.set_enabled_uvm(False)
# [Allocate Tensor2]
...
```
Because this seemingly reads a global setting to change the behavior of destructor, I am afraid that it can be broken if the setting is changed after it is constructed.

Reply:
Not yet. We can add some state to get that to work, but we'd really like to be able to control it at the tensor level if possible, allowing for some tensors as managed, others not. That may solve the problem above in a more holistic way. We'd love to collaborate on that.